### PR TITLE
[ENH] Savitzky-Golay Transformer

### PR DESCRIPTION
**Pull request by Cyril-Meyer**
#### Reference Issues/PRs

Implement Savitzky-Golay filter asked in #1920.

#### What does this implement/fix? Explain your changes.

Add a new transformation, named `SGSeriesTransformer` which use [SciPy implementation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.savgol_filter.html)

#### Does your contribution introduce a new dependency? If yes, which one?

No, but we add another SciPy usage.

#### Any other comments?

Code is based on previous work `DFTSeriesTransformer`. I don't think I've forgotten anything, but there may still be references to the DFT as I started by copying _dft.py and test_dft.py.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [x] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [x] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

### CODE REVIEW

**Comment by Cyril-Meyer** [comment]
Here is a visualization of x12r vs fit_transform(x12r) with the tests parameters.

| v window_length / polyorder > | 2 | 3 | 4 |
| :-: | :-: | :-: | :-: |
| 5 | SG_5_2 | SG_5_3 | SG_5_4 |
| 9 | SG_9_2 | SG_9_3 | SG_9_4 |
| 17 | SG_17_2 | SG_17_3 | SG_17_4 |

**Comment by MatthewMiddlehurst** [comment]
Hi, sorry it took a while to get around to reviewing these, it's been a busy few weeks and the next two will be as well. I will try get around to reviewing this tommorow.

**Comment by MatthewMiddlehurst** [comment]
Seems to be a conflict currently in the init

**Comment by Cyril-Meyer** [comment]
> Seems to be a conflict currently in the init

Probably coming from #1807, I think I fixed it, but we should wait for test to be ok.


> Hi, sorry it took a while to get around these PRs, it's been a busy few weeks and the next two will be as well. I will try to get around to reviewing this tomorrow.

No problem, we all do the best with the time that is given to us :)

**Review by MatthewMiddlehurst — COMMENTED** [review]
Looks fine for the most part, minor change requested. Any comments @TonyBagnall?

File: aeon/transformations/series/_sg.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+class SGSeriesTransformer(BaseSeriesTransformer):
+    """Filter a times series using Savitzky-Golay (SG).
+
+    Parameters
+    ----------
+    window_length : int
+        The length of the filter window (i.e., the number of coefficients).
+        window_length must be less than or equal to the size of the input.
+
+    polyorder : int
```
[the remark this answers is not in the document]
Comment (Cyril-Meyer) [reply aeon/transformations/series/_sg.py]: Done in 86ff79030350efbedb90853ac88ecc9594f5b842

**Comment by Cyril-Meyer** [comment]
@MatthewMiddlehurst : just so I know for next time, when a change is requested, is it up to me to mark the conversation as resolved or should I not touch it?

File: aeon/transformations/series/_sg.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        X : np.ndarray
+            time series in shape (n_channels, n_timepoints)
+        y : ignored argument for interface compatibility
+
+        Returns
+        -------
+        transformed version of X
+        """
+        # Compute SG
+        X_ = savgol_filter(X, self.window_length, self.polyorder)
```
Comment (TonyBagnall) [inline aeon/transformations/series/_sg.py:73]: do you need to pass the axis to savgol for multivariate case? I think its fine as is, the default is scipy is -1 which for us would be 1 I think?

Comment (Cyril-Meyer) [reply aeon/transformations/series/_sg.py:73]: It depends on what we want to do for multivariate.
For now, I tested that if we have a multivariate series (with axis 0 as the channel) the result for each channel is equivalent to a univariate series (check `x12` assert in the test file).
But we can add an option for user to change it if needed.

**Review by TonyBagnall — APPROVED** [review]
LGTM, thanks
